reana_commons/specification.py

```diff
@@ -94,10 +94,8 @@ def _prepare_kwargs(reana_yaml):
             **_prepare_kwargs(reana_yaml),
         )
 
-        if (
-            workflow_type == "cwl" or workflow_type == "snakemake"
-        ) and "inputs" in reana_yaml:
-            input_file = reana_yaml["inputs"]["parameters"]["input"]
+        input_file = reana_yaml.get("inputs", {}).get("parameters", {}).get("input")
+        if (workflow_type == "cwl" or workflow_type == "snakemake") and input_file:
```

Comment:
Works nicely :+1: 

One minor comment is that since we have `if input_file` here, we could delete it from the `if` statement coming on the next line so that can simply read `if workspace_path` only.

(And, to make things more readable, perhaps reformulate this line `if input_file and workflow_type in ("cwl", "snakemake")` so that it is clear that we are doing this branch only when `input_file` exists.)